### ISSUE TITLE
Add migrate_run instead of migrate_tools

### ISSUE THREAD

**Issue by chx** [drush-ops/drush#2926]
https://www.drupal.org/node/2905741 says "Add Migrate Tools to Drush (contrib)"

Instead add https://www.drupal.org/project/migrate_run

migrate_tools depends on migrate_plus which is an incredibly, incredibly broken module, full of bad ideas and missing understanding on how D8 works in general and how D8 migrate is architected.

**Comment by weitzman** [comment]
I'll take a look. Happy to see some fresh ideas in this area.

**Comment by weitzman** [comment]
I'd love to see more issue queue activity and usage for migrate-run before putting it in drush core.

**Comment by weitzman** [comment]
migrate-run looks great. happy to see drush9 support there.

More clear requirements for putting this in Drush core
1. A maintainer. Ideally @claudiu-cristea keeps maintaining it as part of the Drush team
1. Tests
1. At least one person posts here saying that they've used the commands successfully on a project.

**Comment by claudiu-cristea** [comment]
@weitzman, I agree to maintain it as part of Drush. Related to tests, I guess we should use the Drush testing with unish as we test console commands but how this can be done as this is still a Drupal module?

**Comment by chx** [comment]
Do I count as a person :) ?

I have three projects at least (or is it four now? Lost count) where I used migrate_run successfully.

**Comment by weitzman** [comment]
@chx counts as a small nation 💪. Thanks for the positive report. Thats good enough IMO.

@claudiu-cristea I think you stop being a module and ship these commands in the src/Drupal/Commands/core dir. And yes, Unish tests.

**Comment by claudiu-cristea** [comment]
#3402 is ready for review.

**Comment by claudiu-cristea** [comment]
@weitzman @chx, #3402 is ready but  there’s still one important thing to figure out. Users of `migrate_run` module are covered, as this is a straight port of that module into Drush core. But users of `migrate_tools` that were using the `migrate_plus` way to store migrations as config entities? I guess they are not to many because `migrate_tools` knows also to run migrations stored as simple YAML files (not as config entities). The problem is that till we merge #3402 they're still able run their migrations. But once the Drush core migrate commands are in, buuum! I guess we have to think some kind of BC for them. Few ideas:
- `migrate_tools` is renaming their Drush commands.
- We are renaming the core commands in #3402

**Comment by chx** [comment]
> `migrate_tools` is renaming their Drush commands.

For a little bit of background, alexpott have asked migrations not to be config entities. Once that happened, `migrate_plus` decided to go against the explicit guidance of a core maintainer and still do it. They also have plugins that go against the vision I implemented in the migrate architecture. So: they are very well used to do their own thing. Can't see a problem here, they stay in their own world and rename their commands.

**Comment by weitzman** [comment]
It would be helpful if folks could share their migrate ymls that have been used with migrate_run? Were they Drupal => Drupal upgrades or not? Seems like that use case has more merit for config entities than migrations from 3rd party sources.

@chx - please keep this discussion as positive as possible.

**Comment by chx** [comment]
Yeah, I am using it for Drupal to Drupal not sure why would they be different...? When I want to change a core YAML, I just copy it and change it. https://gist.github.com/chx/329c33825ec08aa222535a7a6bddc3d8  the first is the core file migration with `source_base_path` filled in and a constant added. The second in the gist is a completely custom migration. I find it easy to have them at the same place, easy to edit et al. The way this goes (for  me) by the time we finish editing them, the project is over so having them in files is best.

**Comment by claudiu-cristea** [comment]
I wonder if there are use cases in the real world that are only using the `migrate_drupal` core module without writing or touching any of the migrate YAML file. I mean, you have a Drupal 6 site and you want to migrate it as it is, without changing, improving, adding new features. I'm still managing some D6 sites that will never be changed as features and I will keep them in D6. We would someone migrate a D6/7 site to D8 without changing anything?

However, I'll add some docs on how to run and extend such a migration. Recently I had a big D6 > D8 migration and I kept `migrate_drupal` disabled but I copied and adapted files from `migrate_drupal` in my custom module.

**Comment by weitzman** [comment]
I actually think we should document the 'copy and adapt' approach. I agree that this is much more common.

**Comment by juampynr** [comment]
Thanks for the quick feedback @weitzman. Did you just say that Drush is going into core? If so, where can I read more about this?

**Comment by weitzman** [comment]
Hah, just a typo. Fixed now.
